cni/pkg/install/install.go

```diff
@@ -233,7 +233,7 @@ func checkInstall(cfg *config.InstallConfig, cniConfigFilepath string) error {
 	}
 	defaultCNIConfigFilepath := filepath.Join(cfg.MountedCNINetDir, defaultCNIConfigFilename)
 	if defaultCNIConfigFilepath != cniConfigFilepath {
-		if len(cfg.CNIConfName) > 0 {
+		if len(cfg.CNIConfName) > 0 || !cfg.ChainedCNIPlugin {
```

Comment:
This is not needed, in L227 it is already excluded

Reply:
FYI #38158 added

Reply:
@hzxuzhonghu I want to install the istio plugin. Istio cni plugin is used as a standalone plugin (in other words, cfg.chainedCNIPlugin = false)

Reply:
@hzxuzhonghu if `cfg.CNIEnableInstall` is true, istio will not write the CNI configuration and copy the binary. It is not expected.

Reply:
Can you check L227, this is no needed